editor/state.js

```diff
@@ -37,39 +142,46 @@ export const blocks = combineReducers( {
 		}
 
 		return state;
-	},
-	selected( state = {}, action ) {
-		switch ( action.type ) {
-			case 'TOGGLE_BLOCK_SELECTED':
-				return {
-					...state,
-					[ action.uid ]: action.selected
-				};
-		}
+	}
+}, { resetTypes: [ 'REPLACE_BLOCKS' ] } );
 
-		return state;
-	},
-	hovered( state = {}, action ) {
-		switch ( action.type ) {
-			case 'TOGGLE_BLOCK_HOVERED':
-				return {
-					...state,
-					[ action.uid ]: action.hovered
-				};
+/**
+ * Reducer returning selected block state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Dispatched action
+ * @return {Object}        Updated state
+ */
+export function selectedBlock( state = null, action ) {
+	switch ( action.type ) {
+		case 'TOGGLE_BLOCK_SELECTED':
+			return action.selected ? action.uid : null;
+	}
 
-			case 'TOGGLE_BLOCK_SELECTED':
-				if ( state[ action.uid ] ) {
-					return {
-						...state,
-						[ action.uid ]: false
-					};
-				}
-				break;
-		}
+	return state;
+}
 
-		return state;
+/**
+ * Reducer returning hovered block state.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Dispatched action
+ * @return {Object}        Updated state
+ */
+export function hoveredBlock( state = null, action ) {
```

Comment:
Let's keep in mind the cognitive overload nd perceived complexity for people whenever we split files. (Part of why redux is perceived as more complex than what it is.)